oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java

```diff
@@ -286,23 +280,31 @@ public void afterBulk(long executionId, BulkRequest bulkRequest, BulkResponse bu
         }
 
         @Override
-        public void afterBulk(long executionId, BulkRequest bulkRequest, Throwable throwable) {
-            LOG.error("ElasticIndex Update Bulk Failure : Bulk with id {} threw an error", executionId, throwable);
-            suppressedExceptions.add(throwable);
+        public void afterBulk(long executionId, BulkRequest request, List<String> contexts, Throwable failure) {
+            LOG.error("ElasticIndex Update Bulk Failure : Bulk with id {} threw an error", executionId, failure);
+            suppressedErrorCauses.add(ErrorCause.of(ec -> {
+                    StringWriter sw = new StringWriter();
+                    PrintWriter pw = new PrintWriter(sw);
+                    failure.printStackTrace(pw);
+                    return ec.reason(failure.getMessage()).stackTrace(sw.toString());
+            }));
             phaser.arriveAndDeregister();
```

Comment:
Shouldn't this call be on a finally block so it is executed even if the callback method fails with an exception? Same comment for the other `afterBulk` method.